src/schema/v2/partnerDocumentsConnection.ts

```diff
@@ -0,0 +1,142 @@
+import { ResolverContext } from "types/graphql"
+import { GravityIDFields } from "./object_identification"
+import { BodyAndHeaders, ResponseHeaders } from "lib/loaders"
+
+export const PartnerDocumentType = new GraphQLObjectType<any, ResolverContext>({
+  name: "PartnerDocument",
+  fields: {
+    ...GravityIDFields,
+    title: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    filesize: {
+      type: new GraphQLNonNull(GraphQLInt),
+      resolve: ({ size }) => size,
+    },
+    size: {
+      type: new GraphQLNonNull(GraphQLInt),
+      deprecationReason: "Prefer `filesize`",
+    },
+    publicURL: {
+      type: new GraphQLNonNull(GraphQLString),
+      resolve: ({ uri }) => `${config.GRAVITY_API_BASE}/${uri}`,
+    },
+    publicUrl: {
+      type: new GraphQLNonNull(GraphQLString),
+      deprecationReason: "Prefer `publicURL`",
+      resolve: ({ uri }) => `${config.GRAVITY_API_BASE}/${uri}`,
+    },
+  },
+})
+interface GravityArgs {
+  document_ids?: string[]
+  offset: number
+  size: number
+  total_count: boolean
+}
+
+const FILTER_KEYS = ["documentIDs", "artistID", "showID"]
+
+export const PartnerDocumentsConnection: GraphQLFieldConfig<
+  any,
+  ResolverContext
+> = {
+  description: "Return partner documents if current user has CMS access.",
+  type: connectionWithCursorInfo({
+    nodeType: PartnerDocumentType,
+  }).connectionType,
+  args: pageable({
+    documentIDs: {
+      type: new GraphQLList(GraphQLString),
+      description: "Filter documents by ID.",
+    },
+    artistID: {
+      type: GraphQLString,
+    },
+    showID: {
+      type: GraphQLString,
+    },
+  }),
+  resolve: async (
+    { _id: partnerID },
+    args,
+    {
+      partnerDocumentsLoader,
+      partnerArtistDocumentsLoader,
+      partnerShowDocumentsLoader,
+    }
+  ) => {
+    if (
+      !partnerArtistDocumentsLoader ||
+      !partnerArtistDocumentsLoader ||
+      !partnerShowDocumentsLoader
+    ) {
+      return null
+    }
+
+    if (
+      Object.keys(args).filter((key) => FILTER_KEYS.includes(key)).length > 1
+    ) {
+      throw Error("Only one filter arg supported at a time.")
+    }
+    const { page, size, offset } = convertConnectionArgsToGravityArgs(args)
+    const gravityArgs: GravityArgs = {
+      size,
+      offset,
+      total_count: true,
+    }
+
+    let response: BodyAndHeaders<any, ResponseHeaders> = {
+      body: undefined,
+      headers: {},
+    }
+
+    const { artistID, showID, documentIDs } = args
+
+    if (artistID) {
+      response = await partnerArtistDocumentsLoader(
+        { artistID, partnerID },
+        gravityArgs
+      )
+    } else if (showID) {
+      response = await partnerShowDocumentsLoader(
+        { showID, partnerID },
+        gravityArgs
+      )
+    } else {
+      if (documentIDs) {
+        gravityArgs.document_ids = flatten([args.documentIDs])
+      }
+
+      response = await partnerDocumentsLoader!(partnerID, gravityArgs)
+    }
```

Comment:
FWIW: I think this resolver could be simplified quite a bit by adding additional `artist_id` and `show_id` filters to the existing `/api/v1/partner/:id/documents` endpoint. Then we could always use the partnerDocumentsLoader and get ride of the switching altogether.

I don't think we're gaining much benefit from having three different endpoints that have very similar behavior:

- /api/v1/partner/:id//documents ([GitHub](https://github.com/artsy/gravity/blob/0c439fed720e6add00fb802fa10881d246a56872/app/api/v1/partners_documents_endpoint.rb))
- /api/v1/partner/:id/artist/:artist_id/documents ([GitHub](https://github.com/artsy/gravity/blob/0c439fed720e6add00fb802fa10881d246a56872/app/api/v1/partners_shows_documents_endpoint.rb))
- /api/v1/partner/:id/show/:show_id/documents ([GitHub](https://github.com/artsy/gravity/blob/0c439fed720e6add00fb802fa10881d246a56872/app/api/v1/partners_artists_documents_endpoint.rb))